tests/nfs_claim_released_pv.go

```diff
@@ -0,0 +1,317 @@
+/*
+Copyright 2021 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tests
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/ghodss/yaml"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	mayav1alpha1 "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	deploy "github.com/openebs/dynamic-nfs-provisioner/pkg/kubernetes/api/apps/v1/deployment"
+	container "github.com/openebs/dynamic-nfs-provisioner/pkg/kubernetes/api/core/v1/container"
+	pvc "github.com/openebs/dynamic-nfs-provisioner/pkg/kubernetes/api/core/v1/persistentvolumeclaim"
+	pts "github.com/openebs/dynamic-nfs-provisioner/pkg/kubernetes/api/core/v1/podtemplatespec"
+	volume "github.com/openebs/dynamic-nfs-provisioner/pkg/kubernetes/api/core/v1/volume"
+	provisioner "github.com/openebs/dynamic-nfs-provisioner/provisioner"
+)
+
+var _ = Describe("TEST RECLAIM OF RELEASED NFS PV", func() {
+	var (
+		// application parameters
+		applicationNamespace = "default"
+		appName              = "busybox-nfs"
+		appLabel             = "demo=nfs-deployment"
+		appLabelSelector     = map[string]string{
+			"demo": "nfs-deployment",
+		}
```

Comment:
nit/optional: Good to have busybox-deployment as nfs-deployment may leads to little confusion
```suggestion
		appLabel             = "demo=busybox-deployment"
		appLabelSelector     = map[string]string{
			"demo": "busybox-deployment",
		}
```

Reply:
done